tests/e2e/rollouts_test.go

```diff
@@ -286,10 +212,10 @@ var _ = Describe("RolloutManager tests", func() {
 
 				By("updating the deployment when the image in the RolloutManager is updated")
 				Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(&rolloutManager), &rolloutManager)).To(Succeed())
-				rolloutManager.Spec = rolloutsmanagerv1alpha1.RolloutManagerSpec{
-					Image:   controllers.DefaultArgoRolloutsImage,
-					Version: controllers.DefaultArgoRolloutsVersion,
-				}
+
+				rolloutManager.Spec.Image = controllers.DefaultArgoRolloutsImage
+				rolloutManager.Spec.Version = controllers.DefaultArgoRolloutsVersion
+
 				Expect(k8sClient.Update(ctx, &rolloutManager)).To(Succeed())
```

Comment:
It will use the same function as this https://github.com/argoproj-labs/argo-rollouts-manager/pull/39#discussion_r1550776597. Should we remove retry mechanism from controller but still use it for tests?

Reply:
Yah, my suggestion is to remove the retry mechanism from the controller, but keep it for the tests:
- The Reconcile() loop is a standard operator pattern, where, if an error occurs, the resource will be re-reconciled and the logic will run again. This necessarily means *re-running* some of the logic and API calls again, but this is very little work: API calls are just simple GET requests from the operator to etcd, and the logic that runs is trivial.
    - Ultimately the goal is, with each Reconcile run, the resource gets closer to it's desired, declaratively defined state.
- In contrast, for the tests, we don't have an equivalent to Reconcile mechanism, so we need to use our own retry loop.